src/Form/OverrideEntityForm.php

```diff
@@ -0,0 +1,223 @@
+
+    $definitions = $this->entityFieldManager->getFieldDefinitions($entity->getEntityTypeId(), $entity->bundle());
+    $options = $definitions[$field_name]->getSetting('overwritable_properties')[$referenced_entity->bundle()]['options'];
+
+    foreach ($form_display->getComponents() as $name => $component) {
+      if (!in_array($name, array_filter($options))) {
+        $form_display->removeComponent($name);
+      }
+    }
+
+    $form_display->buildForm($referenced_entity, $form, $form_state);
+
+    $form['actions'] = ['#type' => 'actions'];
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Save'),
+      '#button_type' => 'primary',
+    ];
+
+    $form_state->set('entity_reference_override', [
+      'entity_type' => $entity_type,
+      'entity_id' => $entity_id,
+      'field_name' => $field_name,
+      'delta' => $delta,
+    ]);
+
+    if ($this->isAjax()) {
+      $form['actions']['submit']['#ajax']['callback'] = '::ajaxSubmit';
+    }
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    if ($this->isAjax()) {
+      return;
+    }
+
+    $arguments = $form_state->get('entity_reference_override');
+
+    /** @var \Drupal\Core\Entity\FieldableEntityInterface $entity */
+    $entity = $this->entityTypeManager->getStorage($arguments['entity_type'])
+      ->load($arguments['entity_id']);
+
+    $entity->{$arguments['field_name']}->get($arguments['delta'])->overwritten_property_map = $this->getOverwrittenValues($form_state, $entity);
+    $entity->save();
+
+  }
+
+  /**
+   * Get overwritten values for an entity.
+   *
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state object.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The parent entity.
+   *
+   * @return string
+   *   The overridden values as JSON.
+   */
+  protected function getOverwrittenValues(FormStateInterface $form_state, EntityInterface $entity) {
+    $arguments = $form_state->get('entity_reference_override');
+
+    /** @var \Drupal\Core\Entity\EntityInterface $referenced_entity */
+    $referenced_entity = $entity->{$arguments['field_name']}->get($arguments['delta'])->entity;
+
+    $definitions = $this->entityFieldManager->getFieldDefinitions($entity->getEntityTypeId(), $entity->bundle());
+
+    $options = $definitions[$arguments['field_name']]->getSetting('overwritable_properties')[$referenced_entity->bundle()]['options'];
+
+    $values = [];
+    foreach ($options as $name => $enabled) {
+      if ($enabled) {
+        $values = [$name => $form_state->getValue($name)];
+      }
+    }
+    return Json::encode($values);
+  }
+
+  /**
+   * The access function.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  public static function access() {
+    return AccessResult::allowed();
```

Comment:
Do we need a follow-up to make this check edit access to the field?

Reply:
Opened https://github.com/thunder/entity_reference_override/issues/8